p2p_wallet/Scenes/CreateOrRestoreWallet/Common/Pincode/Pincode/PincodeViewController.swift

```diff
@@ -31,29 +32,76 @@ final class PincodeViewController: BaseViewController {
         BESafeArea {
             BEVStack {
                 UIImageView(
-                    width: 160,
-                    height: 120,
+                    width: 114,
+                    height: 107,
                     image: UIImage.lockPincode,
                     contentMode: .scaleAspectFit
                 )
-                    .padding(.init(top: 24, left: .zero, bottom: 24, right: .zero))
+                    .padding(.init(
+                        top: 70 * UIScreen.main.bounds.height / 812,
+                        left: .zero,
+                        bottom: 35,
+                        right: .zero
+                    ))
+                UIView.spacer
 
                 UILabel(
                     font: .font(of: .title2, weight: .regular),
                     textAlignment: .center
                 )
                     .bind(titleLabel)
 
+                BEVStack {
+                    PinCode(correctPincode: viewModel.pincode, bottomLeftButton: self.bottomLeftButton())
+                        .setup { view in
+                            view.stackViewSpacing = 34
+                            view.resetingDelayInSeconds = 1
+                        }
+                        .bind(pincodeView)
+                        .padding(.init(
+                            top: 0,
+                            left: .zero,
+                            bottom: 41,
+                            right: .zero
+                        ))
+                    UIView.spacer
+                    if viewModel.showForgetPin {
+                        forgetPinView()
+                    }
+                }
+                .frame(height: viewModel.showForgetPin ? 417 : 417 - 24)
```

Comment:
do we need to hardcore height?

Reply:
didn't find better option to keep numpad size